python/ml4ir/base/features/feature_fns/sequence.py

```diff
@@ -0,0 +1,57 @@
+import tensorflow as tf
+from tensorflow.keras import layers
+from tensorflow import io
+
+
+def bytes_sequence_to_encoding(feature_tensor, feature_info):
+    """
+    Encode a string tensor into an encoding.
+    Works by converting the string into a bytes sequence and then generating
+    a categorical/char embedding for each of the 256 bytes. The char/byte embeddings
+    are then combined using a biLSTM
+
+    Args:
+        feature_tensor: String feature tensor that is to be encoded
+        feature_info: Dictionary representing the feature_config for the input feature
+
+    Returns:
+        Encoded feature tensor
+
+    Args under feature_layer_info:
+        max_length: int; max length of bytes sequence
+        embedding_size: int; dimension size of the embedding;
+                        if null, then the tensor is just converted to its one-hot representation
+        encoding_size: int: dimension size of the sequence encoding computed using a biLSTM
+
+    NOTE:
+        The input dimension for the embedding is fixed to 256 because the string is
+        converted into a bytes sequence.
+    """
+    feature_layer_info = feature_info["feature_layer_info"]
+
+    # Decode string tensor to bytes
+    feature_tensor = io.decode_raw(
+        feature_tensor,
+        out_type=tf.uint8,
+        fixed_length=feature_layer_info["args"].get("max_length", None),
+    )
+
+    feature_tensor = tf.reshape(feature_tensor, [-1, feature_layer_info["args"]["max_length"]])
+    if "embedding_size" in feature_layer_info["args"]:
+        char_embedding = layers.Embedding(
+            input_dim=256,
+            output_dim=feature_layer_info["args"]["embedding_size"],
+            mask_zero=True,
+            input_length=feature_layer_info["args"]["max_length"],
+        )(feature_tensor)
+    else:
+        char_embedding = tf.one_hot(feature_tensor, depth=256)
+
+    encoding = layers.Bidirectional(
```

Comment:
Are we explicitly assuming here that the user wants a character-level BiLSTM? If so, would it make sense to call the fn CharacterLevelBiLSTMEncoder (the name smells like a class, I get it)? With high probability I am missing context here, so feel free to disregard...

Reply:
No. You are right. Initially, I had the `bilstm` part as an argument so the user could control how you encode the sequence through the feature config, but then removed it because it was probably too generic. I think calling this a bilstm encoder is a great idea. I will do that.